packages/core-mobile/app/new/routes/onboarding/keystone/createPin.tsx

```diff
@@ -0,0 +1,55 @@
+import { useStoredBiometrics } from 'common/hooks/useStoredBiometrics'
+import { useRouter } from 'expo-router'
+import { CreatePin as Component } from 'features/onboarding/components/CreatePin'
+import { useWallet } from 'hooks/useWallet'
+import React, { useCallback } from 'react'
+import { useSelector } from 'react-redux'
+import AnalyticsService from 'services/analytics/AnalyticsService'
+import { WalletType } from 'services/wallet/types'
+import { selectActiveWalletId } from 'store/wallet/slice'
+import BiometricsSDK from 'utils/BiometricsSDK'
+import Logger from 'utils/Logger'
+import { uuid } from 'utils/uuid'
+
+export default function CreatePin(): JSX.Element {
+  const { navigate } = useRouter()
+  const { onPinCreated } = useWallet()
+  const { isBiometricAvailable, useBiometrics, setUseBiometrics } =
+    useStoredBiometrics()
+  const activeWalletId = useSelector(selectActiveWalletId)
+
+  const handleEnteredValidPin = useCallback(
+    (pin: string): void => {
+      AnalyticsService.capture('OnboardingPasswordSet')
+      onPinCreated({
+        walletId: activeWalletId ?? uuid(),
+        mnemonic: uuid(),
+        pin,
+        walletType: WalletType.KEYSTONE
+      })
+        .then(() => {
+          if (useBiometrics) {
+            BiometricsSDK.enableBiometry().catch(Logger.error)
+          }
+          navigate({
+            // @ts-ignore TODO: make routes typesafe
+            pathname: '/onboarding/keystone/setWalletName'
+          })
+        })
```

Comment:
this part seems to not match the latest code. could you resync with main and adjust it to match other createPin screens?